### PR TITLE
[tests-only][full-ci] adding test for editing shared file(OCM) by the share receiver

### PR DESCRIPTION
**Pull request by ghost**
## Description

adding test for editing shared file(OCM) by the share receiver
bug coverage of https://github.com/owncloud/ocis/issues/10358

## Related Issue

- Related Issue https://github.com/owncloud/ocis/issues/10073

## How Has This Been Tested?

- test environment:
- test case 1:
- test case 2:
- ...

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: tests/acceptance/features/apiOcm/share.feature [hunk 1]
```diff
@@ -618,3 +618,24 @@ Feature: an user shares resources using ScienceMesh application
...
+    And user "Alice" has uploaded file with content "ocm test" to "/textfile.txt"
+    And user "Alice" has sent the following resource share invitation to federated user:
+      | resource        | textfile.txt                  |
+      | space           | Personal                      |
+      | sharee          | Brian                         |
+      | shareType       | user                          |
+      | permissionsRole | File Editor                   |
+      | federatedServer | @federation-ocis-server:10200 |
+    And using server "REMOTE"
+    And for user "Brian" the content of the file "/textfile.txt" of the space "Shares" should be "ocm test"
```
Comment (saw-jan) [inline tests/acceptance/features/apiOcm/share.feature:638]: Please, refer to https://github.com/owncloud/ocis/issues/10536
and you will need to do something like https://github.com/owncloud/ocis/blob/3ec2a4e695327ed444d4280d30c761b11418c7a2/tests/acceptance/bootstrap/SpacesTUSContext.php#L153-L156 done (WIP) in https://github.com/owncloud/ocis/pull/10415

File: tests/acceptance/features/apiOcm/share.feature [hunk 2]
```diff
@@ -733,3 +733,26 @@ Feature: an user shares resources using ScienceMesh application
     When using server "LOCAL"
     And user "Alice" uploads a file with content "lorem" to "file.txt" inside federated share "FOLDER" via TUS using the WebDAV API
     Then for user "Alice" the content of file "file.txt" of federated share "FOLDER" should be "lorem"
+
+  @issue-9898
+  Scenario: user edists content of a federated share file
```
Comment (PrajwolAmatya) [inline tests/acceptance/features/apiOcm/share.feature:738]:
```suggestion
  Scenario: user edits content of a federated share file
```

File: tests/acceptance/bootstrap/SpacesContext.php [hunk 3]
```diff
@@ -1345,6 +1345,29 @@ public function theUserUploadsAFileToSpace(
...
+	 * @param string $user
+	 * @param string $spaceName
+	 * @param string $content
+	 * @param string $destination
+	 *
+	 * @return void
+	 * @throws GuzzleException
+	 * @throws Exception
+	 */
+	public function theUserUploadsAFileToFedratedSpace(
```
Comment (S-Panta) [inline tests/acceptance/bootstrap/SpacesContext.php:1360]:
```suggestion
	public function theUserUploadsAFileInsideFedratedSpace(
```
Just matching step definition

File: tests/acceptance/bootstrap/SpacesContext.php [hunk 4]
```diff
@@ -1345,6 +1345,29 @@ public function theUserUploadsAFileToSpace(
...
+		string $user,
+		string $spaceName,
+		string $content,
+		string $destination
+	): void {
+		$spaceId = $this->featureContext->spacesContext->getSharesRemoteItemId($user, $destination);
+		$response = $this->featureContext->uploadFileWithContent($user, $content, $destination, $spaceId);
+		$this->featureContext->setResponse($response);
```
Comment (amrita-shrestha) [inline tests/acceptance/bootstrap/SpacesContext.php:1368]: we may could this too, bt not compulsary
```suggestion
		$this->featureContext->setResponse($this->featureContext->uploadFileWithContent($user, $content, $destination, $spaceId));
```

File: tests/acceptance/features/apiOcm/share.feature [hunk 5]
```diff
@@ -733,3 +733,26 @@ Feature: an user shares resources using ScienceMesh application
...
+    Given using spaces DAV path
```
Comment (amrita-shrestha) [inline tests/acceptance/features/apiOcm/share.feature:739]: if we need this step in all scenario we can put it in backgroung

Comment (amrita-shrestha) [inline tests/acceptance/features/apiOcm/share.feature:739]: if we need this step in all scenario we can put it in backgroung

File: tests/acceptance/bootstrap/SpacesContext.php (hunk 4 above)
Comment (saw-jan) [reply tests/acceptance/bootstrap/SpacesContext.php]: the existing one is cleaner. let's try to split if the line gets more detailed

Comment (saw-jan) [reply tests/acceptance/bootstrap/SpacesContext.php]: or better formatting would do

File: tests/acceptance/features/apiOcm/share.feature [hunk 6]
```diff
@@ -733,3 +733,26 @@ Feature: an user shares resources using ScienceMesh application
...
+    And user "Alice" has uploaded file with content "ocm test" to "/textfile.txt"
+    And user "Alice" has sent the following resource share invitation to federated user:
+      | resource        | textfile.txt |
+      | space           | Personal     |
+      | sharee          | Brian        |
+      | shareType       | user         |
+      | permissionsRole | File Editor  |
+    And using server "REMOTE"
+    And for user "Brian" the content of file "textfile.txt" of federated share "textfile.txt" should be "ocm test"
+    When user "Brian" uploads a file inside federated space Shares with content "this is a new content" to "textfile.txt" using the WebDAV API
```
Comment (saw-jan) [inline tests/acceptance/features/apiOcm/share.feature:754]:
```suggestion
    When user "Brian" updates the content of federated share "textfile.txt" with "this is a new content" using the WebDAV API
```

File: tests/acceptance/bootstrap/SpacesContext.php [hunk 7]
```diff
@@ -1345,6 +1345,33 @@ public function theUserUploadsAFileToSpace(
...
+	 * @param string $destination
+	 *
+	 * @return void
+	 * @throws GuzzleException
+	 * @throws Exception
+	 */
+	public function theUserUploadsAFileInsideFedratedSpace(
+		string $user,
+		string $content,
+		string $destination
```
Comment (saw-jan) [inline tests/acceptance/bootstrap/SpacesContext.php:1362]:
```suggestion
		string $share
```

File: tests/acceptance/bootstrap/SpacesContext.php [hunk 8]
```diff
@@ -1345,6 +1345,33 @@ public function theUserUploadsAFileToSpace(
...
+		string $content,
+	): void {
+		$spaceId = $this->featureContext->spacesContext->getSharesRemoteItemId($user, $share);
+		$this->featureContext->setResponse(
+			$this->featureContext->uploadFileWithContent(
+				$user,
+				$content,
+				$share,
+				$spaceId
+			)
```
Comment (saw-jan) [inline tests/acceptance/bootstrap/SpacesContext.php:1371]: what happens with the share param in this function? is it used in the request?

File: tests/acceptance/bootstrap/SpacesContext.php [hunk 9]
```diff
@@ -1345,6 +1345,33 @@ public function theUserUploadsAFileToSpace(
...
+	 * @return void
+	 * @throws GuzzleException
+	 * @throws Exception
+	 */
+	public function userUpdatesTheContentOfFederatedShareWithUsingTheWebdavApi(
+		string $user,
+		string $share,
+		string $content,
+	): void {
+		$spaceId = $this->featureContext->spacesContext->getSharesRemoteItemId($user, $share);
```
Comment (PrajwolAmatya) [inline tests/acceptance/bootstrap/SpacesContext.php:1364]:
```suggestion
		$spaceId = $this->getSharesRemoteItemId($user, $share);
```

File: tests/acceptance/bootstrap/SpacesContext.php (hunk 8 above)
Comment (saw-jan) [reply tests/acceptance/bootstrap/SpacesContext.php:1371]: this should not be triggered, right? because we send some value to `$spaceId`

Comment (saw-jan) [reply tests/acceptance/bootstrap/SpacesContext.php:1371]: I see this url being hit:
`https://localhost:10200/remote.php/dav/spaces/89f37a33-858b-45fa-8890-a1f2b27d90e1$f1c7b05d-38c4-4822-b30c-7c484b8ac287!L3RleHRmaWxlLnR4dA==/textfile.txt`
but this should work
`https://localhost:10200/remote.php/dav/spaces/89f37a33-858b-45fa-8890-a1f2b27d90e1$f1c7b05d-38c4-4822-b30c-7c484b8ac287!L3RleHRmaWxlLnR4dA==/`

you can send the empty string in that param

**Review by saw-jan — APPROVED** [review]

**Review by amrita-shrestha — APPROVED** [review]

**Pull request merged**